client/ayon_speedtree/api/pipeline.py

```diff
@@ -0,0 +1,326 @@
+    etc.
+
+    Args:
+        context (dict): context data
+
+    """
+    return write_context_metadata(SPTREE_SECTION_NAME_CONTEXT, context)
+
+
+def write_context_metadata(metadata_key, context):
+    """Write context data into the related json
+    which stores in .sptree_metadata/key folder
+    in the project work directory.
+
+    The context data includes the project name, folder path
+    and task name.
+
+    Args:
+        metadata_key (str): metadata key
+        context (dict): context data
+    """
+    work_dir = get_workdir()
+    json_dir = os.path.join(
+        work_dir, ".zbrush_metadata", metadata_key).replace(
+            "\\", "/"
+        )
+    os.makedirs(json_dir, exist_ok=True)
+    json_file = f"{json_dir}/{metadata_key}.json"
+    if os.path.exists(json_file):
+        with open (json_file, "r") as file:
+            value = json.load(file)
+            if value == context:
+                return
+    with open (json_file, "w") as file:
+        value = json.dumps(context)
+        file.write(value)
+        file.close()
+
+
+def get_current_workfile_context():
+    """Function to get the current context data from the related
+    json file in .sptree_metadata/context folder
+
+    The current context data includes thing like project name,
+    folder path and task name.
+
+    Returns:
+        list: list of context data
+    """
+    return get_load_context_metadata()
+
+
+def get_load_context_metadata():
+    """Get the context data from the related json file
+    ("context.json") which stores in .sptree_metadata/context
+    folder in the project work directory.
+
+    The context data includes the project name, folder path and
+    task name.
+
+    Returns:
+        list: context data
+    """
+    file_content = {}
+    work_dir = get_workdir()
+    json_dir = os.path.join(
+        work_dir, ".sptree_metadata", SPTREE_SECTION_NAME_CONTEXT).replace(
+            "\\", "/"
+        )
+    if not os.path.exists(json_dir):
+        return file_content
+    file_list = os.listdir(json_dir)
+    if not file_list:
+        return file_content
+    for file in file_list:
+        with open (f"{json_dir}/{file}", "r") as data:
+            content = ast.literal_eval(str(data.read().strip()))
+            file_content.update(content)
+            data.close()
+    return file_content
+
+
+
+def set_current_file(filepath=None):
+    """Function to store current workfile path
+
+    Args:
+        filepath (str, optional): current workfile path. Defaults to None.
+    """
+    work_dir = get_workdir()
+    txt_dir = os.path.join(
+        work_dir, ".sptree_metadata").replace(
+            "\\", "/"
+    )
+    os.makedirs(txt_dir, exist_ok=True)
+    txt_file = f"{txt_dir}/current_file.txt"
+    if filepath is None:
+        with open(txt_file, "w"):
+            pass
+        return filepath
+    filepath_check = tmp_current_file_check()
+    if filepath_check.endswith("spm"):
```

Comment:
What's this check needed for?

Reply:
No need to check, it should be the temp file store somewhere instead.